### PR TITLE
Support parse proto-text format http request body

### PR DESCRIPTION
**Pull request by hiberabyss**

### CODE REVIEW

File: src/brpc/policy/http_rpc_protocol.cpp [hunk 1]
```diff
@@ -513,6 +522,12 @@ void SerializeHttpRequest(butil::IOBuf* /*not used*/,
                 return cntl->SetFailed(EREQUEST, "Fail to serialize %s",
                                        pbreq->GetTypeName().c_str());
             }
+        } else if (content_type == HTTP_CONTENT_PROTO_TEXT) {
+            if (!google::protobuf::TextFormat::Print(*pbreq, &wrapper)) {
+                cntl->request_attachment().clear();
+                return cntl->SetFailed(EREQUEST, "Fail to print %s",
```
Comment (wwbmmm) [inline src/brpc/policy/http_rpc_protocol.cpp:528]: Fail to print %s as proto-text

File: src/brpc/policy/http_rpc_protocol.cpp [hunk 2]
```diff
@@ -757,6 +772,10 @@ HttpResponseSender::~HttpResponseSender() {
             if (!res->SerializeToZeroCopyStream(&wrapper)) {
                 cntl->SetFailed(ERESPONSE, "Fail to serialize %s", res->GetTypeName().c_str());
             }
+        } else if (content_type == HTTP_CONTENT_PROTO_TEXT) {
+            if (!google::protobuf::TextFormat::Print(*res, &wrapper)) {
+                cntl->SetFailed(ERESPONSE, "Fail to print %s", res->GetTypeName().c_str());
```
Comment (wwbmmm) [inline src/brpc/policy/http_rpc_protocol.cpp:777]: Fail to print %s as proto-text

File: src/brpc/policy/http_rpc_protocol.cpp [hunk 3]
```diff
@@ -1470,6 +1489,12 @@ void ProcessHttpRequest(InputMessageBase *msg) {
                                     req->GetDescriptor()->full_name().c_str());
                     return;
                 }
+            } else if (content_type == HTTP_CONTENT_PROTO_TEXT) {
+                if (!ParsePbTextFromIOBuf(req, req_body)) {
+                    cntl->SetFailed(EREQUEST, "Fail to parse http body as %s",
```
Comment (wwbmmm) [inline src/brpc/policy/http_rpc_protocol.cpp:1494]: Fail to parse http proto-text body as %s

**Comment by wwbmmm** [comment]
@hiberabyss Thanks for your contribution! Can you add some unit test for this feature?

File: src/brpc/policy/http_rpc_protocol.cpp (hunk 1 above)
Comment (hiberabyss) [reply src/brpc/policy/http_rpc_protocol.cpp]: Done

File: src/brpc/policy/http_rpc_protocol.cpp (hunk 2 above)
Comment (hiberabyss) [reply src/brpc/policy/http_rpc_protocol.cpp]: Done

File: src/brpc/policy/http_rpc_protocol.cpp (hunk 3 above)
Comment (hiberabyss) [reply src/brpc/policy/http_rpc_protocol.cpp]: Done

**Comment by hiberabyss** [comment]
> @hiberabyss Thanks for your contribution! Can you add some unit test for this feature?

It's a little complicated to run test on M1 macOS, could this MR be merged first? Have tested this with my own service.

**Comment by wwbmmm** [comment]
> It's a little complicated to run test on M1 macOS

brpc has not support M1 macOS yet, you can test it on a linux device. You can follow the compiling and testing guide here: https://github.com/apache/incubator-brpc/blob/master/docs/cn/getting_started.md

**Comment by hiberabyss** [comment]
> @hiberabyss Thanks for your contribution! Can you add some unit test for this feature?

FYI, add some ut for proto-text.

File: test/brpc_http_rpc_protocol_unittest.cpp [hunk 4]
```diff
@@ -310,6 +325,12 @@ TEST_F(HttpTest, verify_request) {
         VerifyMessage(msg, false);
         msg->Destroy();
     }
+    {
+        brpc::policy::HttpContext* msg =
+                MakePostProtoTextRequestMessage("/EchoService/Echo");
+        VerifyMessage(msg, false);
+        msg->Destroy();
```
Comment (wwbmmm) [inline test/brpc_http_rpc_protocol_unittest.cpp:332]: I seems that this unit test does not cover any code you change in `http_rpc_protocol.cpp`. You can reference the unit test named `spring_protobuf_content_type` in `brpc_http_rpc_protocol_unittest.cpp` which also test a new content type.

Comment (hiberabyss) [reply test/brpc_http_rpc_protocol_unittest.cpp:332]: add new spring_protobuf_text_content_type ut.

Comment (hiberabyss) [reply test/brpc_http_rpc_protocol_unittest.cpp:332]: Sorry, forgot to push the commit. Now it should be there.

Comment (wwbmmm) [reply test/brpc_http_rpc_protocol_unittest.cpp:332]: Good job!

**Pull request merged**